.github/workflows/release.yml

```diff
@@ -0,0 +1,132 @@
+name: 'release'
+
+on:
+  push:
+    tags: [ '*' ]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+    steps:
+      - uses: actions/checkout@v2
+        id: code-checkout
+
+      - name: Validate composer.json and composer.lock
+        id: composer-validate
+        run: composer validate
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-php-
+
+      - name: Install dependencies
+        if: steps.composer-cache.outputs.cache-hit != 'true'
+        id: install-dependencies
+        uses: php-actions/composer@v5
+        with:
+          php_version: 8.1
+          args: --prefer-dist --no-progress --no-suggest
+
+      - name: PHP Unit Test
+        id: php-codesniffer
+        run: vendor/bin/phpunit
+
+#      - uses: act10ns/slack@v1
+#        with:
+#          status: ${{ job.status }}
+#          steps: ${{ toJson(steps) }}
+#          channel: '#wordpress-bmlt-workflow'
+#        if: failure()
+
+  package:
+    runs-on: ubuntu-latest
+   # needs: [ lint ] # tmp let job run if tests fail
```

Comment:
I currently have this commented out till tests can be fixed, when uncommented  this makes the build stage only run if tests pass